src/shared/components/Pagination/Pagination.tsx

```diff
@@ -16,16 +18,39 @@ export type PaginationProps = {
 export const Pagination: React.FC<PaginationProps> = ({
   itemsPerPage = 0,
   totalCount = 0,
-  maxPaginationLinks = 5,
+  defaultPaginationLength = 5,
   page = 0,
   onChangePage,
 }) => {
+  const [paginationLength, setPaginationLength] = useState(defaultPaginationLength)
+
+  const calculatePaginationLength = () => {
+    const arrowsWithMargins = PAGINATION_BUTTON_WIDTH + ARROWS_MARGINS * 4
+    const viewPortDifference =
+      window.innerWidth - VIEWPORT_PADDING - (PAGINATION_BUTTON_WIDTH * defaultPaginationLength + arrowsWithMargins)
+    if (viewPortDifference < 0) {
+      setPaginationLength(defaultPaginationLength - Math.ceil(Math.abs(viewPortDifference) / PAGINATION_BUTTON_WIDTH))
+    } else {
+      setPaginationLength(defaultPaginationLength)
+    }
+  }
```

Comment:
I'm not a fan of the amount of variables we need to take into account here. I think we could make `Pagination` take 100% width and run resize observer on it. Then we don't need to bother with viewport size/padding